### ISSUE TITLE
Start to use AdaDelta as optimizer

### ISSUE THREAD

**Issue by davidsandberg** [davidsandberg/facenet#4]
Start to use AdaDelta as optimizer when it's available in tensorflow
https://github.com/tensorflow/tensorflow/pull/644

**Comment by davidsandberg** [comment]
Currently AdaGrad with a learning rate lr=0.1 is used. This seems to work fairly well without putting any effort into learning rate optimization.

**Comment by davidsandberg** [comment]
The possibility to use Adadelta optimizer has been added. With learning_rate=0.1, rho=0.9, epsilon=1e-6 the training gets significantly slower. Need to look more into this.

**Comment by davidsandberg** [comment]
Did another attempt with the AdaDelta optimizer. It seems to work fairly well but no notable difference to the AdaGrad optimizer. More details can be found on the wiki: https://github.com/davidsandberg/facenet/wiki/Performance-optimization
